core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -160,12 +161,14 @@ public ClusterState execute(ClusterState currentState) {
                 MetaData.Builder mdBuilder = MetaData.builder(currentState.metaData());
                 ClusterBlocks.Builder blocksBuilder = ClusterBlocks.builder()
                         .blocks(currentState.blocks());
+                Version minIndexCompatibilityVersion = currentState.getNodes().getSmallestNonClientNodeVersion()
```

Comment:
the actual question is why do we use the smallest, we have to use the largest node in the cluster since that is the one relevant for the index version to be opened.